src/components/QuranReader/PlayChapterAudioButton.tsx

```diff
@@ -1,58 +1,54 @@
 import { useContext } from 'react';
 
+import { useSelector } from '@xstate/react';
 import useTranslation from 'next-translate/useTranslation';
-import { useDispatch, useSelector, shallowEqual } from 'react-redux';
 
 import PauseIcon from '../../../public/icons/pause.svg';
 import PlayIcon from '../../../public/icons/play-arrow.svg';
-import { triggerPauseAudio, triggerPlayAudio } from '../AudioPlayer/EventTriggers';
 
 import styles from './PlayButton.module.scss';
 
 import Button, { ButtonSize, ButtonType, ButtonVariant } from 'src/components/dls/Button/Button';
 import DataContext from 'src/contexts/DataContext';
-import useGetQueryParamOrReduxValue from 'src/hooks/useGetQueryParamOrReduxValue';
-import {
-  exitRepeatMode,
-  playFrom,
-  selectAudioData,
-  selectIsPlaying,
-  selectIsRadioMode,
-  selectPlaybackRate,
-} from 'src/redux/slices/AudioPlayer/state';
 import { getChapterData } from 'src/utils/chapter';
 import { logButtonClick } from 'src/utils/eventLogger';
-import QueryParam from 'types/QueryParam';
+import { AudioPlayerMachineContext } from 'src/xstate/AudioPlayerMachineContext';
 
 interface Props {
   chapterId: number;
 }
 const PlayChapterAudioButton: React.FC<Props> = ({ chapterId }) => {
   const { t } = useTranslation('common');
-  const dispatch = useDispatch();
-  const isAudioPlaying = useSelector(selectIsPlaying);
-  const playbackRate = useSelector(selectPlaybackRate);
-  const currentAudioData = useSelector(selectAudioData, shallowEqual);
-  const isRadioMode = useSelector(selectIsRadioMode);
   const chaptersData = useContext(DataContext);
   const chapterData = getChapterData(chaptersData, chapterId.toString());
 
-  const isPlayingCurrentChapter = isAudioPlaying && currentAudioData?.chapterId === chapterId;
+  const audioService = useContext(AudioPlayerMachineContext);
+  const isAudioPlaying = useSelector(audioService, (state) =>
+    state.matches('VISIBLE.AUDIO_PLAYER_INITIATED.PLAYING'),
+  );
+  const currentSurah = useSelector(audioService, (state) => state.context.surah);
+  const isPlayingCurrentChapter = isAudioPlaying && currentSurah === chapterId;
 
-  const { value: reciterId }: { value: number } = useGetQueryParamOrReduxValue(QueryParam.Reciter);
   const play = () => {
     logButtonClick('chapter_header_play_audio');
-    dispatch(exitRepeatMode());
-    if (currentAudioData?.chapterId === chapterId && !isRadioMode) {
-      triggerPlayAudio(playbackRate);
+    const { surah } = audioService.getSnapshot().context;
+
+    // if surah is being played, toggle it
+    if (Number(chapterId) === surah) {
```

Comment:
But if it's being played already and the user clicks play chapter again, it will be paused this way if we are dispatching `TOGGLE` event.